### PR TITLE
Change verbosity arguments from the base command to the upgrade related commands

### PR DESCRIPTION
**Pull request by drehak**
Change verbosity arguments from the base command to the upgrade related commands

The verbose and debug options it's now only available in commands where make sense to
increase verbosity.

A re-request of https://github.com/oamg/leapp/pull/574 by @alexandrepossebom, since he left.

### CODE REVIEW

File: leapp/cli/__init__.py [hunk 1]
```diff
@@ -7,14 +7,8 @@
...
-@command_opt('debug', is_flag=True, help='Enable debug mode', inherit=True)
-@command_opt('verbose', is_flag=True, help='Enable verbose logging', inherit=True)
-def cli(args):
-    os.environ['LEAPP_DEBUG'] = '1' if args.debug else os.environ.get('LEAPP_DEBUG', '0')
-    if os.environ['LEAPP_DEBUG'] == '1' or args.verbose:
-        os.environ['LEAPP_VERBOSE'] = '1'
-    else:
-        os.environ['LEAPP_VERBOSE'] = os.environ.get('LEAPP_VERBOSE', '0')
+def cli(args): # noqa; pylint: disable=unused-argument
+    pass
```
Comment (pirat89) [inline leapp/cli/__init__.py]: re-raising original post: @vinzenz, do you know how to get rid of this function here? I haven't studied the cli code much but it seems to me it has no meaning here now.

Comment (bocekm) [reply leapp/cli/__init__.py]: @vinzenz, could you please add a comment explaining why this function needs to be here. Thanks. We can just leave this strange code here but it should be at least properly commented.

Comment (vinzenz) [reply leapp/cli/__init__.py]: this function used to handle debug and verbose for all sub commands by setting the environment variable, so not every sub command had to implement this.

Comment (vinzenz) [reply leapp/cli/__init__.py]: and i am not sure what is so strange about it, it was just a way of not having to do the same stuff every time

File: leapp/cli/upgrade/__init__.py [hunk 2]
```diff
@@ -198,13 +210,14 @@ def upgrade(args):
...
-             help='Enables experimental actors')
+@command_opt('whitelist-experimental', action='append', metavar='ActorName', help='Enables experimental actors')
 @command_opt('save-answerfile', help='Path to save custom answerfile')
+@command_opt('debug', is_flag=True, help='Enable debug mode', inherit=False)
+@command_opt('verbose', is_flag=True, help='Enable verbose logging', inherit=False)
 def preupgrade(args):
     if os.getuid():
         raise CommandError('This command has to be run under the root user.')
-
+    parse_verbose(args)
```
Comment (vinzenz) [inline leapp/cli/upgrade/__init__.py]: you're not really parsing it, but handling it

Comment (vinzenz) [reply leapp/cli/upgrade/__init__.py]: also it handles debug too, so this function name is kind of misleading

File: leapp/cli/__init__.py (hunk 1 above)
Comment (bocekm) [reply leapp/cli/__init__.py]: I understand the original purpose, but now it has lost it. I and Alex were trying to get rid of it but with no success.

Comment (drehak) [reply leapp/cli/__init__.py]: Alright, we'll keep it this way.

Comment (vinzenz) [reply leapp/cli/__init__.py]: > ... with no success.

could you elaborate on that?

File: leapp/cli/upgrade/__init__.py (hunk 2 above)
Comment (drehak) [reply leapp/cli/upgrade/__init__.py]: Good points, replaced with `handle_output_level`.

**Comment by drehak** [comment]
Applied feedback and squashed.

File: leapp/cli/__init__.py (hunk 1 above)
Comment (bocekm) [reply leapp/cli/__init__.py]: > could you elaborate on that?
I don't remember what constructs I was trying but basically whatever change I tried to remove the function warranted a traceback. After I gave up for not wielding enough python powers.

**Pull request merged**